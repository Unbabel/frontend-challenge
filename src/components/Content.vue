<template>
    <section class="content">
        <Transition>
            <TransitionGroup v-if="!listIsEmpty" tag="ul" class="list">
                <li v-for="note in notes" :key="note.id">
                    <Note :data="note" />
                </li>
            </TransitionGroup>
        </Transition>
        <button @click="addNote" class="btn" :aria-label="$t('aria.add')">
            <Icon name="add" />
        </button>
    </section>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { mapMutations, mapState } from 'vuex'
import Note from '@/components/Note.vue'
import Icon from '@/components/icons/Icon.vue'

export default defineComponent({
    components: {
        Note,
        Icon
    },
    computed: {
        ...mapState(['notes']),
        listIsEmpty() {
            return this.notes.length === 0
        }
    },
    methods: {
        ...mapMutations(['addNote'])
    }
})
</script>

<style src="@/assets/css/transitions.scss" lang="scss" scoped />
<style lang="scss" scoped>
$header: 69px;
$footer: 36px;

.content {
    min-height: calc(100% - $header - $footer);
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-start;
}

.list {
    width: 100%;
    list-style: none;
    padding: 0;
    margin: 0;

    @media(min-width: 768px) {
        margin-top: 24px;
        max-width: 738px;
    }
}

.btn {
    margin: 24px 0;
}
</style>