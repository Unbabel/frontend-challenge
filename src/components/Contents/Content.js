import * as React from 'react'

const Content = ({ value, inputType, onChange, onBlur, placeholder }) => {
    return (inputType === "single" ? < input value={value}
        onChange={onChange}
        onBlur={onBlur}
        autoFocus={true}
        placeholder={placeholder}
    />
        :
        <textarea value={value}
            onChange={onChange}
            onBlur={onBlur}
            autoFocus={true}
            placeholder={placeholder}
        />
    )
}
export default Content;