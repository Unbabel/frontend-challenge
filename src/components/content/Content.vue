<script setup></script>

<template>
  <div>
    <slot />
  </div>
</template>

<style scoped></style>
