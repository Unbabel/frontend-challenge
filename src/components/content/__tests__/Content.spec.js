import { describe, it, expect, beforeEach } from "vitest";

import { mount } from "@vue/test-utils";

// Components
import Content from "../Content.vue";

describe("Content", () => {
	let wrapper;

	beforeEach(() => {
		// Arrange
		wrapper = mount(Content);
	});

	it("matches snapshot", () => {
		// Assert
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("renders properly", () => {
		// Act
		const div = wrapper.find("div");

		// Assert
		expect(div.exists()).toBe(true);
	});
});
