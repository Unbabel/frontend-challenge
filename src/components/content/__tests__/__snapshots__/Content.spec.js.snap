// Vitest Snapshot v1

exports[`Content > matches snapshot 1`] = `"<div data-v-c3eac7d6=\\"\\"></div>"`;
